src/wallet/test/coinselector_tests.cpp

```diff
@@ -922,6 +922,169 @@ BOOST_AUTO_TEST_CASE(effective_value_test)
+                CoinsResult available_coins;
+                for (int j = 0; j < 400; ++j) {
+                    add_coin(available_coins, wallet, CAmount(0.0625 * COIN), CFeeRate(0), 144, false, 0, true);
+                }
+                for (int j = 0; j < 2000; ++j) {
+                    add_coin(available_coins, wallet, CAmount(0.025 * COIN), CFeeRate(0), 144, false, 0, true);
+                }
+                return available_coins;
+            },
+            chain, m_args);
+
+        const bool has_0_0625_coin = std::any_of(result->GetInputSet().begin(), result->GetInputSet().end(), [](const auto& coin) {
+            return coin.GetEffectiveValue() == 0.0625 * COIN;
+        });
+        const bool has_0_025_coin = std::any_of(result->GetInputSet().begin(), result->GetInputSet().end(), [](const auto& coin) {
+            return coin.GetEffectiveValue() == 0.025 * COIN;
+        });
+        const bool has_not_50_coin = std::any_of(result->GetInputSet().begin(), result->GetInputSet().end(), [](const auto& coin) {
+            return coin.GetEffectiveValue() != 50 * COIN;
+        });
+        BOOST_CHECK(has_0_0625_coin);
+        BOOST_CHECK(has_0_025_coin);
+        BOOST_CHECK(has_not_50_coin);
+    }
+
+    {
+        // Scenario 3 (available as RPC test):
+
+        // The actor starts with 1x 50.0 BTC, 400x 0.0625 BTC and 2000x 0.025 BTC (125.0 BTC total) unspent outputs
+        // Then tries to spend 49.5 BTC
+        // A combination of coins should be selected, such that the created transaction is not too large
+
+        // Perform selection
+        const auto result = select_coins(
+            target, cs_params, cc, [&](CWallet& wallet) {
+                CoinsResult available_coins;
+                for (int j = 0; j < 400; ++j) {
+                    add_coin(available_coins, wallet, CAmount(0.0625 * COIN), CFeeRate(0), 144, false, 0, true);
+                }
+                for (int j = 0; j < 2000; ++j) {
+                    add_coin(available_coins, wallet, CAmount(0.025 * COIN), CFeeRate(0), 144, false, 0, true);
+                }
+                add_coin(available_coins, wallet, CAmount(50 * COIN), CFeeRate(0), 144, false, 0, true);
+                return available_coins;
+            },
+            chain, m_args);
+
+        // Perform selection
+        const bool has_0_025_coin = std::any_of(result->GetInputSet().begin(), result->GetInputSet().end(), [](const auto& coin) {
+            return coin.GetEffectiveValue() == 0.025 * COIN;
+        });
+        const bool has_0_0625_coin = std::any_of(result->GetInputSet().begin(), result->GetInputSet().end(), [](const auto& coin) {
+            return coin.GetEffectiveValue() == 0.0625 * COIN;
+        });
+        const bool has_not_50_coin = std::any_of(result->GetInputSet().begin(), result->GetInputSet().end(), [](const auto& coin) {
+            return coin.GetEffectiveValue() != 50 * COIN;
+        });
```

Comment:
In fc5219f51813b6e9c128d6bc0cc70c63a5edce48 "test: Check max transaction weight in CoinSelection"

It's not clear to my why we shouldn't expect the 50 Coin to be picked? ISTM Random selection should eventually find the 50 Coin single input solution and go with that.

Reply:
Right, I removed the check. The two other coins should always be present though.